pychiquito/dsl.py

```diff
@@ -82,10 +83,21 @@ def pragma_disable_q_enable(self: Circuit) -> None:
 
     def add(self: Circuit, step_type: StepType, args: Any):
         assert self.mode == CircuitMode.Trace
+        if self.num_step_instances >= self.ast.num_steps:
+            raise ValueError(f"Number of step instances exceeds {self.ast.num_steps}")
+        self.num_step_instances += 1
```

Comment:
@qwang98 isn' this the lenght of the self.witness.step_instances array? I don't think we need to keep self.num_step_instances.

Reply:
Deleted this field and use len(self.witness.step_instances) instead! Thanks. :)